src/key.rs

```diff
@@ -86,35 +88,38 @@ pub trait DcKey: Serialize + Deserializable + KeyTrait + Clone {
     }
 }
 
-#[async_trait]
 impl DcKey for SignedPublicKey {
     type KeyType = SignedPublicKey;
 
-    async fn load_self(context: &Context) -> Result<Self::KeyType> {
-        let addr = context.get_primary_self_addr().await?;
-        match context
-            .sql
-            .query_row_optional(
-                r#"
-            SELECT public_key
-              FROM keypairs
-             WHERE addr=?
-               AND is_default=1;
-            "#,
-                paramsv![addr],
-                |row| {
-                    let bytes: Vec<u8> = row.get(0)?;
-                    Ok(bytes)
-                },
-            )
-            .await?
-        {
-            Some(bytes) => Self::from_slice(&bytes),
-            None => {
-                let keypair = generate_keypair(context).await?;
-                Ok(keypair.public)
+    fn load_self<'a>(
+        context: &'a Context,
+    ) -> Pin<Box<dyn Future<Output = Result<Self::KeyType>> + 'a + Send>> {
+        Box::pin(async move {
+            let addr = context.get_primary_self_addr().await?;
+            match context
+                .sql
+                .query_row_optional(
+                    r#"
+                    SELECT public_key \
+                      FROM keypairs \
+                     WHERE addr=? \
+                       AND is_default=1;\
```

Comment:
why switch to using raw string literals _plus_  backslashes? imu, one should be enough.

Reply:
The reasoning was that I indented the lines with more spaces, and I didn't want to make the SQL statements longer because they have to be parsed, idk if that makes a difference.

However, I just noticed that you can't use raw strings together with backslashes (it's reassuring though that this made like six dozen tests fail)